### PR TITLE
refactor: User module APIs.

### PR DESCRIPTION
**Pull request by pranalidhanavade**
- Refactored following APIs:
1. GET `/users/platform-settings` Get all platform and ecosystem settings.
2. PUT `/users/platform-settings` Update platform and ecosystem settings.
3. GET `/users/user-credentials` Get user credentials by Id.
4. POST `/users/certificate` Share user certificate

### What ###
1. Ensured consistency of API status codes.
2. Modified the NATS function.
5. Tested with both positive and negative scenarios.

### How ###
1. Checked and standardized API status codes to ensure a uniform response.
2. Modified the NATS function to accommodate the adjustments made in the API.

### Why ###
1. Ensuring consistent API status codes improves communication between different components of the system and enhances overall reliability.

### Extra points ###

- Implementation of well-defined interface by replacing previous existing Promise<object> is remaining for this APIs.

### CODE REVIEW

File: libs/common/src/response-messages/index.ts [hunk 1]
```diff
@@ -18,7 +18,8 @@ export const ResponseMessages = {
             userCredentials: 'User credentials fetched successfully',
             platformEcosystemettings: 'Platform and ecosystem settings updated',
             fetchPlatformSettings: 'Platform settings fetched',
-            signUpUser:'User created successfully'
+            signUpUser:'User created successfully',
+            shareUserCertificate:'Certificate url generated successfully'
```
Comment (vivekayanworks) [inline libs/common/src/response-messages/index.ts:22]: URL should be in capital letters.

File: apps/user/interfaces/user.interface.ts [hunk 2]
```diff
@@ -113,10 +113,10 @@ export interface ISendVerificationEmail {
   }
   
   export interface ICheckUserDetails {
```
Comment (vivekayanworks) [inline apps/user/interfaces/user.interface.ts:115]: Can we reduce no. of fields as discussed earlier?

File: libs/common/src/response-messages/index.ts (hunk 1 above)
Comment (pranalidhanavade) [reply libs/common/src/response-messages/index.ts:22]: Resolved.

File: apps/user/interfaces/user.interface.ts (hunk 2 above)
Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Yes sir.

Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Resolved.

**Review by vivekayanworks — APPROVED** [review]

**Pull request merged**